src/packagedcode/podspec.py

```diff
@@ -0,0 +1,279 @@
+            license_type = re.sub(r"/*.*license.*= ", "", line)
+            sub_license = re.sub(r",.*", "", license_type)
+            stripped_license = sub_license.replace("'", "")
+            stripped_license = stripped_license.replace('"', "")
+            stripped_license = stripped_license.replace('{', "")
+            stripped_license = stripped_license.replace('}', "")
+            stripped_license = stripped_license.replace(':type =>', "")
+            stripped_license = stripped_license.strip()
+            podspec_data['license'] = stripped_license
+        elif '.author' in line:
+            authors = re.sub(r"/*.*author.*= ", "", line)
+            stripped_authors = authors.replace("'", "")
+            stripped_authors = stripped_authors.replace('"', "")
+            stripped_authors = stripped_authors.replace('{', "")
+            stripped_authors = stripped_authors.replace('}', "")
+            stripped_authors = stripped_authors.strip()
+            stripped_authors = stripped_authors.split(', ')
+            author_names = []
+            author_email = []
+            for split_author in stripped_authors:
+                split_author = split_author.split('=>')
+                author = split_author[0].strip()
+                email = split_author[1].strip()
+                author_names.append(author)
+                author_email.append(email)
+            podspec_data['author'] = author_names
+            podspec_data['email'] = author_email
+        elif '.summary' in line:
+            summary = re.sub(r"/*.*summary.*= ", "", line)
+            stripped_summary = summary.replace("'", "")
+            stripped_summary = stripped_summary.replace('"', "")
+            stripped_summary = stripped_summary.replace('{', "")
+            stripped_summary = stripped_summary.replace('}', "")
+            stripped_summary = stripped_summary.strip()
+            podspec_data['summary'] = stripped_summary
+        elif '.description' in line:
+            podspec_data['description'] = get_description(location)
+        elif '.homepage' in line:
+            homepage_url = re.sub(r"/*.*homepage.*= ", "", line)
+            stripped_homepage_url = homepage_url.replace("'", "")
+            stripped_homepage_url = stripped_homepage_url.replace('"', "")
+            stripped_homepage_url = stripped_homepage_url.replace('%q{', "")
+            stripped_homepage_url = stripped_homepage_url.replace('}', "")
+            stripped_homepage_url = stripped_homepage_url.strip()
+            podspec_data['homepage_url'] = stripped_homepage_url
+        elif '.dependency' in line:
+            match = add_dependency_regex.match(line)
+            line = match.group('line')
+            podspec_data['dependencies'] = parse_line(line)
+
+    deps = OrderedDict()
+    dependency = podspec_data.get('dependencies')
+    if dependency:
+        for dep in dependency:
+            dep_name = dep[0]
+            dep.pop(0)
```

Comment:
What is this?

Reply:
dependency list contains dep like `["name", "version1", "version2", ....]`. pop deletes the name after storing name into veriable and rest are assigned to version variable.

Reply:
OK, this should be explained in the code.
Also the general structure of the code could be simplified:
```
    dependency = podspec_data.get('dependencies', []) or []
```
would guarantee you always get a list, so you no longer need the indented `if dependency:` guard.
Also please use plural names for collections (dependencies not dependency)

Also I would likely always check the length of a sequence before playing with indices and there are way to make this more explicit and avoid deleting list elements:

```Python
    dependencies = podspec_data.get('dependencies', []) or []
    for dependency in dependencies:
        if not dependency:
            continue
        # each dependency consist of a name as a first item and versions afterwards
        # for instance: .......
        name, versions = dependency[0], dependency[1:]
```